bundles/automation/org.eclipse.smarthome.automation.rest/src/main/java/org/eclipse/smarthome/automation/rest/internal/TemplateResource.java

```diff
@@ -48,13 +52,16 @@
  */
 @Path("templates")
 @Api("templates")
+@Component(service = { RESTResource.class, TemplateResource.class })
```

Comment:
Why do you need to export the the implementation class itself as service (has not been done before)?

Reply:
I changed it because of this [comment](https://github.com/eclipse/smarthome/pull/5265#discussion_r188225858)

Reply:
Sorry for the confusion. @maggu2810 is completely right, the REST services should be declared to provide only the RESTResource interface. Please update accordingly @lpapazow. Thanks.

Reply:
Sorry to comment so late, but I would actually prefer if you could remove all changes to the automation engine from this PR, because it will cause huge conflicts for #4468, which is already addressing these components as well.

Reply:
If I want to keep the LocaleService logic, I should injected the locale into the automation RESTResources. Would it be okay to add refererences in the .xml files in the OSGI-INF folder?

Reply:
Oh, I didn't look closely enough to notice that it all is linked to the LocaleService...
No, changing the XML won't be any better wrt conflicts. I guess it is then ok to leave this PR as it is and rather rebase #4468 later on and sync the changes again. @adimova do you agree?